quimb/gen/operators.py

```diff
@@ -343,17 +343,63 @@ def fsim(theta, phi, dtype=complex, **kwargs):
     a = cos(theta)
     b = -1j * sin(theta)
     c = exp(-1j * phi)
-
     gate = [[1, 0, 0, 0],
             [0, a, b, 0],
             [0, b, a, 0],
             [0, 0, 0, c]]
 
+
+
     gate = qu(gate, dtype=dtype, **kwargs)
     make_immutable(gate)
     return gate
 
 
+#https://arxiv.org/pdf/2010.07965.pdf, Eq. 18
+def fsimg( theta, Zeta, chi, gamma, phi, dtype=complex, **kwargs):
```

Comment:
Can you add the `lru_cache` decorator here? - see other functions around. because we make the arrays immutable we can cache them for the same parameters. Since there are five parameters you might want a reasonably large cache size like 2**8.